include/arch/riscv/common/linker.ld

```diff
@@ -108,6 +108,15 @@ SECTIONS
 		KEEP(*(.vectors.*))
     } GROUP_LINK_IN(ROMABLE_REGION)
 
+    SECTION_PROLOGUE(rom_start,,)
```

Comment:
I like the way you added this but believe it needs to be the first section after "_image_rom_start = ROM_BASE" in order to guarantee the exact same location of this section for two different images.  I may be wrong but believe this should go before _VECTOR_SECTION_NAME

Reply:
I tried to build the code with the section in both places. Where you have it works and where I suggested it does not.  Will the vectors always be the same length or will it take setting .= to a constant from ROM_BASE to guarantee this is always in the same location?

Reply:
If we want to rom_start to be the first section, we also need to contains contents of the vector section. (contain at least a branch instruction at reset vector/offsest 0 which jumping to __initialize. And set mtvec to __irq_wrapper at the beginning of __initialize).

Since vector section have been implemented and the contents should not be changed. We should be able to get the same location of rom_start section for two different images even if rom_start section is after vector section.

Reply:
I think being after the vectors is safe. I don't see the length of the vectors changing on a specific chip.